lib/rules/named-grid-areas-no-invalid/__tests__/index.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const { messages, ruleName } = require('..');
+
+testRule({
+	ruleName,
+
+	config: [true],
+
+	accept: [
+		{
+			code: 'div { grid-template-areas: "a a a" "b b b"; }',
+		},
+		{
+			code: `div { grid-template-areas: "head head"
+"nav  main"
+"nav  foot";}`,
+		},
+		{
+			code: `div { grid-template-areas: /* "" */ "head head"
+"nav  main" /* "a b c" */
+"nav  foot" /* "" */;}`,
+		},
+		{
+			code: 'div { grid-template-areas: none; }',
+		},
+	],
+
+	reject: [
+		{
+			code: 'div { grid-template-areas: "a a a" "b b"; }',
+			message: messages.notRectangular(),
+			line: 1,
+			column: 7,
+		},
+		{
+			code: 'div { grid-template-areas: "a a a" "b b a"; }',
+			message: messages.notContiguousOrRectangular(['a']),
+			line: 1,
+			column: 7,
+		},
+		{
+			code: `div { grid-template-areas: "header header header header"
+"main main . sidebar"
+"footer footer footer header"; }`,
+			message: messages.notContiguousOrRectangular(['header']),
+			line: 1,
+			column: 7,
+		},
+		{
+			code: 'div { grid-template-areas: ""; }',
+			message: messages.noRows(),
+			line: 1,
+			column: 7,
+		},
+	],
+});
```

Comment:
Can we add two positions tests, e.g.:

```css
a { grid-template-areas: /* comment */ ""; }
```

```css
a { grid-template-areas: 
   ""; }
```

We'll likely need to make use of the value parser's `sourceIndex` value to get the correct positions.

The [`alpha-value-notation` rule](https://github.com/stylelint/stylelint/blob/master/lib/rules/alpha-value-notation/index.js) has an example of this.

Reply:
Do tests need to use only `a` selector as well, if the selector is not being tested (like in this rule)?